### ISSUE TITLE
using localhost geth but "sync can take longer when using remote eth1 nodes"

### ISSUE THREAD

**Issue by newearthmartin** [sigp/lighthouse#2452]
## Description

Lighthouse says `sync can take longer when using remote eth1 nodes, service: slot_notifier` even though I'm running lighthouse and geth on the same computer.

## Version

v1.4.0-3b600ac

## Present Behaviour

i'm running lighthouse with argument `--eth1-endpoints http://127.0.0.1:8545`. However in the logs it says it is going slower because I'm using a remote eth1 node

```
Jul 13 11:12:17.001 INFO Syncing                                 est_time: 1 day 1 hrs, speed: 17.33 slots/sec, distance: 1583955 slots (31 weeks 2 days), peers: 21, service: slot_notifier
Jul 13 11:12:17.001 WARN Syncing eth1 block cache                est_blocks_remaining: initializing deposits, msg: sync can take longer when using remote eth1 nodes, service: slot_notifier
Jul 13 11:12:29.000 INFO Syncing                                 est_time: 1 day 8 hrs, speed: 13.33 slots/sec, distance: 1583828 slots (31 weeks 2 days), peers: 22, service: slot_notifier
Jul 13 11:12:29.000 WARN Syncing eth1 block cache                est_blocks_remaining: initializing deposits, msg: sync can take longer when using remote eth1 nodes, service: slot_notifier
Jul 13 11:12:41.000 INFO Syncing                                 est_time: 1 day 5 hrs, speed: 14.67 slots/sec, distance: 1583765 slots (31 weeks 2 days), peers: 21, service: slot_notifier
Jul 13 11:12:41.001 WARN Syncing eth1 block cache                est_blocks_remaining: initializing deposits, msg: sync can take longer when using remote eth1 nodes, service: slot_notifier
Jul 13 11:12:53.000 INFO Syncing                                 est_time: 4 days 13 hrs, speed: 4.00 slots/sec, distance: 1583766 slots (31 weeks 2 days), peers: 24, service: slot_notifier
Jul 13 11:12:53.001 WARN Syncing eth1 block cache                est_blocks_remaining: initializing deposits, msg: sync can take longer when using remote eth1 nodes, service: slot_notifier
Jul 13 11:13:05.000 INFO Syncing                                 est_time: 4 days 13 hrs, speed: 4.00 slots/sec, distance: 1583767 slots (31 weeks 2 days), peers: 24, service: slot_notifier
Jul 13 11:13:05.001 WARN Syncing eth1 block cache                est_blocks_remaining: initializing deposits, msg: sync can take longer when using remote eth1 nodes, service: slot_notifier
```
Maybe the message means something else?
Note: The geth node is not fully synced, but near (I didnt run it in a few days). 

## Expected Behaviour

The eth1 node is not remote so the message should be different.

**Comment by paulhauner** [comment]
It's not simple for Lighthouse to determine if you're using a remote node, unfortunately. It might *seem* so, but there's complications with fallback nodes.

I feel you that this confusing though, so I created #2453 to remove the message completely.

**Comment by newearthmartin** [comment]
I just see that lighthouse is throwing an error at the beginning: `Eth1 node returned error: `[...] `chain not synced beyond EIP-155 replay-protection fork block`.

```
Jul 13 23:58:43.011 INFO Lighthouse started                      version: Lighthouse/v1.4.0-3b600ac
Jul 13 23:58:43.011 INFO Configured for network                  name: mainnet
Jul 13 23:58:43.011 INFO Data directory initialised              datadir: /home/martin/.eth2_lighthouse
Jul 13 23:58:43.019 INFO Deposit contract                        address: 0x00000000219ab540356cbb839cbe05303d7705fa, deploy_block: 11184524
Jul 13 23:58:44.838 INFO Hot-Cold DB initialized                 split_state: 0xd9819a91155d194724696dfe86a80269668e0945cba147092c71579b3a6ec064, split_slot: 483744, version: 3, service: freezer_db
Jul 13 23:58:46.057 INFO Starting beacon chain                   method: resume, service: beacon
Jul 13 23:58:57.370 INFO Block production enabled                method: json rpc via http, endpoints: ["http://127.0.0.1:8545/"]
Jul 13 23:58:58.287 INFO Beacon chain initialized                head_slot: 483808, head_block: 0x1b04�461e, head_state: 0x8bcd�9113, service: beacon
Jul 13 23:58:58.288 INFO Timer service started                   service: node_timer
Jul 13 23:58:58.288 INFO UPnP Attempting to initialise routes    service: UPnP
Jul 13 23:58:58.290 INFO Libp2p Service                          peer_id: 16Uiu2HAmRc7iECEZawrSaJdfzkvAyNf4PndthAhPnLLunnqrh6Qm, service: libp2p
Jul 13 23:58:58.291 INFO ENR Initialised                         tcp: Some(9000), udp: Some(9000), ip: Some(190.16.187.218), id: 0x42b9..dc40, seq: 2, enr: enr:-LK4QC032WlK_a2LW-bJ_Zqpvyy8yOEibLX5Ewr5YdLA78AkMjIqMSYWGsd7GjAWkNOVcm6jLDMNq8i58rvkcwidckICh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhL4Qu9qJc2VjcDI1NmsxoQPAa4K_YWOmSHcTSeRd4q4T-8bpyh2EQg0MIA_Pp5XTJoN0Y3CCIyiDdWRwgiMo, service: libp2p
Jul 13 23:58:58.292 INFO Listening established                   address: /ip4/0.0.0.0/tcp/9000/p2p/16Uiu2HAmRc7iECEZawrSaJdfzkvAyNf4PndthAhPnLLunnqrh6Qm, service: libp2p
Jul 13 23:58:58.315 WARN Error connecting to eth1 node endpoint  action: trying fallbacks, endpoint: http://127.0.0.1:8545/, service: eth1_rpc
Jul 13 23:58:58.316 CRIT Couldn't connect to any eth1 node. Please ensure that you have an eth1 http server running locally on http://localhost:8545 or specify one or more (remote) endpoints using `--eth1-endpoints <COMMA-SEPARATED-SERVER-ADDRESSES>`. Also ensure that `eth` and `net` apis are enabled on the eth1 http server, warning: BLOCK PROPOSALS WILL FAIL WITHOUT VALID, SYNCED ETH1 CONNECTION, service: eth1_rpc
Jul 13 23:58:58.316 ERRO Failed to update eth1 cache             error: Failed to update Eth1 service: "All fallback errored: http://127.0.0.1:8545/ => EndpointError(RequestFailed(\"Eth1 node returned error: {\\\"code\\\":-32000,\\\"message\\\":\\\"chain not synced beyond EIP-155 replay-protection fork block\\\"}\"))", retry_millis: 60000, service: eth1_rpc
```

So it may be that this message means something else in reality. I wouldn't just remove the message.

**Comment by michaelsproul** [comment]
The error about not being synced past the EIP-155 block is unrelated (https://github.com/sigp/lighthouse/pull/2304), and is to be expected when Geth is early in its sync.

The Lighthouse warning you saw was _always_ shown when syncing the block cache, and we added it provide a bit of context. Seeing as it's potentially misleading, and making it more accurate would be difficult, we decided to remove it.

**Comment by newearthmartin** [comment]
Ah then its correct to remove the message. Otherwise you get the impression that you have configured something wrongly.

**Comment by newearthmartin** [comment]
While we are at it, another misleading log is this one about not connecting to the eth1 endpoint. It's not that "it couldn't connect" but that it connected but it's giving an error

```
Jul 13 23:58:58.316 CRIT Couldn't connect to any eth1 node. Please ensure that you have an eth1 http server running locally on http://localhost:8545 or specify one or more (remote) endpoints using `--eth1-endpoints <COMMA-SEPARATED-SERVER-ADDRESSES>`. Also ensure that `eth` and `net` apis are enabled on the eth1 http server, warning: BLOCK PROPOSALS WILL FAIL WITHOUT VALID, SYNCED ETH1 CONNECTION, service: eth1_rpc
Jul 13 23:58:58.316 ERRO Failed to update eth1 cache             error: Failed to update Eth1 service: "All fallback errored: http://127.0.0.1:8545/ => EndpointError(RequestFailed(\"Eth1 node returned error: {\\\"code\\\":-32000,\\\"message\\\":\\\"chain not synced beyond EIP-155 replay-protection fork block\\\"}\"))", retry_millis: 60000, service: eth1_rpc
```

**Comment by paulhauner** [comment]
> another misleading log is this one about not connecting to the eth1 endpoint

Thanks, I've addressed this in https://github.com/sigp/lighthouse/pull/2461.

**Comment by paulhauner** [comment]
Resolved via #2461
